src/app/beer_garden/local_plugins/manager.py

```diff
@@ -277,159 +343,229 @@ def stop_one(
+        elif instance_id is not None:
+            the_runner = self._from_instance_id(instance_id)
+
+        if the_runner is None:
             raise Exception(
-                f"Could not determine runner using runner ID {runner_id} and "
-                f"instance ID {instance_id}"
+                f"Could not determine runner using runner ID ({runner_id}) and "
+                f"instance ID ({instance_id})"
             )
 
         if send_sigterm:
-            runner.term()
+            the_runner.term()
 
-        runner.join(config.get("plugin.local.timeout.shutdown"))
+        the_runner.join(config.get("plugin.local.timeout.shutdown"))
 
-        if runner.is_alive():
-            runner.dead = True
-            runner.kill()
+        if the_runner.is_alive():
+            the_runner.dead = True
+            the_runner.kill()
 
-        runner.stopped = True
-        runner.restart = False
+        the_runner.stopped = True
+        the_runner.restart = False
 
         if remove:
-            self._runners.remove(runner)
+            self._runners.remove(the_runner)
 
-        return runner.state()
+        return the_runner.state()
 
     def stop_all(self) -> None:
-        """Stop all known runners"""
+        """Stop all known runners."""
         return self._stop_multiple(self._runners)
 
-    def scan_path(self, paths: Iterable[str] = None) -> List[Runner]:
-        """Scan a given directory for valid plugins
+    def scan_path(self, paths: Optional[Iterable[Path]] = None) -> List[Runner]:
+        """Scan a directory for valid plugins.
 
-        Note: This scan does not walk the directory tree - all plugins must be
-        in the top level of the given path.
+        Note:
+            This scan does not walk the directory tree--all plugins must be in the
+            top-level of the given path.
 
         Args:
-            paths: The paths to scan. If None will be all subdirectories of the plugin
-            path specified at initialization
+            paths: The paths to scan, Optional. If ``None``, will be all subdirectories
+                of the plugin path specified at initialization
+
+        Raises:
+            TypeError: If `paths` is ``None`` and the plugin path specified at
+                initialization is also ``None``.
 
         Returns:
-            None
+            List of Runners to run the plugins.
         """
-        plugin_paths = paths or self._plugin_path.iterdir()
+        if paths is None and self._plugin_path is None:
+            self.logger.exception("PluginManager.scan_path has no path to scan")
+            return []
 
+        plugin_paths = paths or self._plugin_path.iterdir()
         new_runners = []
 
         try:
             for path in plugin_paths:
                 try:
-                    if path.is_dir() and path not in self.paths():
+                    if self._is_valid_plugin_path(path, self.paths(), self.logger):
```

Comment:
All checks for determining whether a path is a valid plugin directory are factored out into a single function, rather than having some here and some elsewhere.